src/libraries/System.Threading.Tasks.Dataflow/src/Blocks/TransformManyBlock.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,303 @@
+            if (dataflowBlockOptions.SupportsParallelExecution && dataflowBlockOptions.EnsureOrdered)
+            {
+                _reorderingBuffer = new ReorderingBuffer<IEnumerable<TOutput>>(
+                    this, (source, messages) => ((TransformManyBlock<TInput, TOutput>)source)._source.AddMessages(messages));
+            }
+
+            // Create the underlying target and source
+            _target = new TargetCore<TInput>(this,
+                messageWithId => ProcessMessage(transform, messageWithId),
+                _reorderingBuffer, dataflowBlockOptions, TargetCoreOptions.UsesAsyncCompletion);
+
+            // Link up the target half with the source half.  In doing so,
+            // ensure exceptions are propagated, and let the source know no more messages will arrive.
+            // As the target has completed, and as the target synchronously pushes work
+            // through the reordering buffer when async processing completes,
+            // we know for certain that no more messages will need to be sent to the source.
+            _target.Completion.ContinueWith((completed, state) =>
+            {
+                var sourceCore = (SourceCore<TOutput>)state!;
+                if (completed.IsFaulted) sourceCore.AddAndUnwrapAggregateException(completed.Exception!);
+                sourceCore.Complete();
+            }, _source, CancellationToken.None, Common.GetContinuationOptions(), TaskScheduler.Default);
+
+            // It is possible that the source half may fault on its own, e.g. due to a task scheduler exception.
+            // In those cases we need to fault the target half to drop its buffered messages and to release its
+            // reservations. This should not create an infinite loop, because all our implementations are designed
+            // to handle multiple completion requests and to carry over only one.
+            _source.Completion.ContinueWith((completed, state) =>
+            {
+                var thisBlock = ((TransformManyBlock<TInput, TOutput>)state!) as IDataflowBlock;
+                Debug.Assert(completed.IsFaulted, "The source must be faulted in order to trigger a target completion.");
+                thisBlock.Fault(completed.Exception!);
+            }, this, CancellationToken.None, Common.GetContinuationOptions() | TaskContinuationOptions.OnlyOnFaulted, TaskScheduler.Default);
+
+            // Handle async cancellation requests by declining on the target
+            Common.WireCancellationToComplete(
+                dataflowBlockOptions.CancellationToken, Completion, state => ((TargetCore<TInput>)state!).Complete(exception: null, dropPendingMessages: true), _target);
+#if FEATURE_TRACING
+            DataflowEtwProvider etwLog = DataflowEtwProvider.Log;
+            if (etwLog.IsEnabled())
+            {
+                etwLog.DataflowBlockCreated(this, dataflowBlockOptions);
+            }
+#endif
+        }
```

Comment:
It's unfortunate how much duplication there is between the existing shared ctor and this one.  I understand you've done so because of the different builds, but it'd be really nice to find a way where the majority of the duplication could be avoided.

Reply:
I tried multiple approaches to avoid duplication but none of them succeeded 😞
Do you have a specific idea in mind?

Reply:
💡
Can we build the Dataflow component only for `$(NetCoreAppCurrent)` (currently `net5.0`)?

Reply:
I found a possible solution by introducing a shared Initialize method to replace the private constructor.
Better ?